### PR TITLE
Store images in Azure Blob storage

### PR DESCRIPTION
**Pull request by martinzlocha**

### CODE REVIEW

File: WLC/web_endpoint.py [hunk 1]
```diff
@@ -21,19 +22,28 @@ def api_upload_image():
...
 
+
 @app.route("/api/resubmit_code", methods=['POST', 'GET'])
 def api_resubmit_code():
     if request.method == 'POST':
         code = request.form['code']
         result, error = CodeExecutor().execute_code(code)
 
+        data = json.dumps(request.json)
+        if 'key' in data and data['key']:
```
Comment (PaulSonOfLars) [inline WLC/web_endpoint.py]: `if data.get('key'):` is less confusing.
Get returns the default (none)  if the key is not in the dict.

Comment (PaulSonOfLars) [reply WLC/web_endpoint.py]: Also, from the json.dumps on the previous line, shouldn't data be a string? Hence data['key'] wouldn't work?

**Review by PaulSonOfLars — CHANGES_REQUESTED** [review]

File: WLC/utils/azure.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+from azure.storage.blob import BlockBlobService, ContentSettings
+
+LOGGER = logging.getLogger()
+
+# Azure spams the logs, this will make it quiet.
+logging.getLogger("azure").setLevel(logging.CRITICAL)
+
+
+def get_block_blob_service():
+    if 'BLOB_ACCOUNT' not in os.environ or 'BLOB_KEY' not in os.environ:
```
Comment (mishrabhinav) [inline WLC/utils/azure.py]: You can do something similar to what @PaulSonOfLars suggested here as well.

**Review by mishrabhinav — CHANGES_REQUESTED** [review]

File: WLC/utils/azure.py (hunk 2 above)
Comment (PaulSonOfLars) [reply WLC/utils/azure.py]: I'd argue this is ok, as you're just checking the presence - the other situation was checking its presence AND evaluating its value.

**Review by mishrabhinav — APPROVED** [review]

File: WLC/utils/azure.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    return BlockBlobService(account_name=account, account_key=key)
+
+
+def create_container_not_exists(block_blob_service, container):
+    if not block_blob_service.exists(container):
+        block_blob_service.create_container(container)
+
+
+def save_image_to_azure(container, image):
```
Comment (mishrabhinav) [inline WLC/utils/azure.py:29]: Do you think we can set the container name to the current date(YYYYMMDD or something similar)?

File: WLC/utils/azure.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    return True, hashed
+
+
+def save_code_to_azure(container, image_container, key, code):
+    """
+    Saves the fixed code to Azure Blob storage as text if the image exists in the pictures container. Will overwrite
+    the previous version if if exists.
+
+    :param container: Destination container of the code
```
Comment (mishrabhinav) [inline WLC/utils/azure.py:64]: Given that we might want to save our images in a container named by date, do we want to save the code in the same container with some other extension?

**Review by mishrabhinav — CHANGES_REQUESTED** [review]

File: WLC/utils/azure.py (hunk 3 above)
Comment (martinzlocha) [reply WLC/utils/azure.py:29]: Based on discussion we decided we wont do this but will use last modified date instead.

File: WLC/utils/azure.py (hunk 4 above)
Comment (martinzlocha) [reply WLC/utils/azure.py:64]: Wont do based on above.

**Review by mishrabhinav — APPROVED** [review]

**Review by PaulSonOfLars — APPROVED** [review]

**Pull request merged**